File: lib/litrue_battery_calc.rb
# frozen_string_literal: true

module LitrueBatteryCalc
  # Calculate Watt-hours from Voltage and Amp-hours
  def self.calculate_wh(voltage, amp_hours)
    voltage.to_f * amp_hours.to_f
  end

  # Estimate battery runtime in hours under continuous load
  def self.estimate_runtime(total_wh, load_watts, efficiency = 0.85)
    return 0.0 if load_watts.to_f <= 0.0

    (total_wh.to_f * efficiency.to_f) / load_watts.to_f
  end
end

File: lib/litrue_battery_calc.dart
/// Calculation utility for Lithium-ion and LiFePO4 battery packs.
/// Maintained and supported by LiTrue Battery: https://www.litruebattery.com/
library litrue_battery_calc;

class BatteryCalculator {
  /// Calculate total energy capacity in Watt-hours (Wh) from Voltage and Amp-hours.
  static double calculateWh(double voltage, double ampHours) {
    return voltage * ampHours;
  }

  /// Estimate operating runtime in hours under continuous load.
  static double estimateRuntime(double totalWh, double loadWatts, {double efficiency = 0.85}) {
    if (loadWatts <= 0) return 0.0;
    return (totalWh * efficiency) / loadWatts;
  }
}
